libethereum/TransactionQueue.cpp

```diff
@@ -370,13 +372,23 @@ void TransactionQueue::clear()
 
 void TransactionQueue::enqueue(RLP const& _data, h512 const& _nodeId)
 {
+	bool queued = false;
 	{
 		Guard l(x_queue);
 		unsigned itemCount = _data.itemCount();
 		for (unsigned i = 0; i < itemCount; ++i)
+		{
+			if (m_unverified.size() >= c_maxVeririficationQueueSize)
+			{
+				clog(TransactionQueueChannel) << "Transaction verification queue is full. Dropping" << itemCount - i << "transactions";
```

Comment:
For malfromed transactions the peer who sends them will eventually loose rating (and presumably banned) here: https://github.com/ethereum/cpp-ethereum/blob/develop/libethereum/EthereumHost.cpp#L386
For well-formed transactions It is more complicated. For instance an attacker could send a stream of perfectly fine transactions, with high gas price from an account with zero balance. Such transactions will only fail at execution phase.  Would it be correct to lower rating for peers that propagate them? Currently peers propagate transactions without checking if they execute first.